crates/radix-engine-toolkit/src/manifest_analysis/classification/pool_contribution.rs

```diff
@@ -0,0 +1,233 @@
+                    ACCOUNT_WITHDRAW_IDENT
+                    | ACCOUNT_DEPOSIT_IDENT
+                    | ACCOUNT_DEPOSIT_BATCH_IDENT
+                    | ACCOUNT_TRY_DEPOSIT_OR_ABORT_IDENT
+                    | ACCOUNT_TRY_DEPOSIT_BATCH_OR_ABORT_IDENT
+                    | ACCOUNT_CREATE_PROOF_OF_AMOUNT_IDENT
+                    | ACCOUNT_CREATE_PROOF_OF_NON_FUNGIBLES_IDENT
+                    | ACCOUNT_LOCK_FEE_IDENT
+                    | ACCOUNT_LOCK_CONTINGENT_FEE_IDENT
+                    | ACCOUNT_LOCK_FEE_AND_WITHDRAW_IDENT,
+                )
+                | (
+                    Some(GroupedEntityType::AccessControllerEntities(..)),
+                    ACCESS_CONTROLLER_CREATE_PROOF_IDENT,
+                )
+                | (
+                    // All of the pool blueprints have the same name for the
+                    // contribute and redeem methods and I wanted to use a
+                    // constant here so I chose to use the constants for the
+                    // one resource pool blueprint. Again, they're all the
+                    // same string for all of the pool blueprints and we do
+                    // not need to have redundant strings here.
+                    Some(GroupedEntityType::PoolEntities(..)),
+                    ONE_RESOURCE_POOL_CONTRIBUTE_IDENT,
+                ) => true,
+                // Disallowed Invocations
+                (
+                    Some(
+                        GroupedEntityType::IdentityEntities(..)
+                        | GroupedEntityType::PoolEntities(..)
+                        | GroupedEntityType::InternalEntities(..)
+                        | GroupedEntityType::SystemEntities(..)
+                        | GroupedEntityType::ResourceManagerEntities(..)
+                        | GroupedEntityType::PackageEntities(..)
+                        | GroupedEntityType::ValidatorEntities(..)
+                        | GroupedEntityType::AccountEntities(..)
+                        | GroupedEntityType::AccessControllerEntities(..)
+                        | GroupedEntityType::GenericComponentEntities(..)
+                        | GroupedEntityType::AccountLockerEntities(..),
+                    )
+                    | None,
+                    _,
+                ) => false,
+            }
+        }
+        GroupedInstruction::TakeFromWorktopInstructions(
+            TakeFromWorktopInstructions::TakeFromWorktop(..)
+            | TakeFromWorktopInstructions::TakeAllFromWorktop(..),
+        ) => true,
+        // Permitted Instructions
+        GroupedInstruction::ReturnToWorktopInstructions(..)
+        | GroupedInstruction::AssertionInstructions(..)
+        | GroupedInstruction::ProofInstructions(..) => true,
+        // Disallowed Instructions
+        GroupedInstruction::TakeFromWorktopInstructions(
+            TakeFromWorktopInstructions::TakeNonFungiblesFromWorktop(..),
+        )
```

Comment:
This seems quite arbitrary - to simplify things, can we just allow all `TakeFromWorktopInstructions`; and just restrict the withdraws/deposits to fungibles only?

Reply:
Yeah, this is currently redundant actually since all of the classifications have the requirements needed to ensure that this case of the match statement never get hit. I'm in favor of removing this special casing to just act on groups rather than on individual members.